inst/templates/Makevars.in

```diff
@@ -0,0 +1,50 @@
+TARGET_DIR = ./rust/target
+LIBDIR = $(TARGET_DIR)/release
+STATLIB = $(LIBDIR)/lib{{{lib_name}}}.a
+PKG_LIBS = -L$(LIBDIR) -l{{{lib_name}}}
+
+all: C_clean
+
+$(SHLIB): $(STATLIB)
+
+CARGOTMP = $(CURDIR)/.cargo
+VENDOR_DIR = $(CURDIR)/vendor
+
+
+# RUSTFLAGS appends --print=native-static-libs to ensure that 
+# the correct linkers are used. Use this for debugging if need. 
+#
+# CRAN note: Cargo and Rustc versions are reported during
+# configure via tools/msrv.R.
+#
+# When on CRAN, the vendor.tar.xz file is unzipped and used
+# for offline compilation. It is ignored when NOT_CRAN != false
+$(STATLIB):
+
+	# Check if NOT_CRAN is false and unzip vendor.tar.xz if so
+	if [ "$(NOT_CRAN)" = "false" ]; then \
+		if [ -f ./rust/vendor.tar.xz ]; then \
+			tar xf rust/vendor.tar.xz && \
+			mkdir -p $(CARGOTMP) && \
+			cp rust/vendor-config.toml $(CARGOTMP)/config.toml; \
+		fi; \
+	fi
+
+	export CARGO_HOME=$(CARGOTMP) && \
+	export PATH="$(PATH):$(HOME)/.cargo/bin" && \
+	RUSTFLAGS="$(RUSTFLAGS) --print=native-static-libs" cargo build @CRAN_FLAGS@ --lib --release --manifest-path=./rust/Cargo.toml --target-dir $(TARGET_DIR)
+
+	# Always clean up CARGOTMP
+	rm -Rf $(CARGOTMP)
+
+ 	if [ "$(NOT_CRAN)" != "false" ]; then \
+        rm -Rf $(VENDOR_DIR); \
+        rm -Rf $(TARGET_DIR); \
+        rm -Rf $(LIBDIR)/build; \
+    fi
```

Comment:
I think i have this logic backwards....I think we _want_ to remove these directories when `NOT_CRAN="false"`? Either way both `R CMD checks` pass so maybe it isn't needed?